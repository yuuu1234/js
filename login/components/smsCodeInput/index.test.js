import React from 'react';
import { Platform } from 'react-native';
import renderer from 'react-test-renderer';
import SMSCodeInput from './index';

test('User sms Code screen renders correctly', async () => {
  const numberDigits = 4;
  const onCanVerifySMSCode = jest.fn();
  const onCanNotVerifySMSCode = jest.fn();
  const keyboardType = Platform.OS === 'ios' ? 'number-pad' : 'numeric';
  const tree = renderer
    .create(
      <SMSCodeInput
        numberDigits={numberDigits}
        onCanVerifySMSCode={onCanVerifySMSCode}
        onCanNotVerifySMSCode={onCanNotVerifySMSCode}
        keyboardType={keyboardType}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
