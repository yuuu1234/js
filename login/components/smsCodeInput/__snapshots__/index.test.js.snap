// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User sms Code screen renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#003D9A",
        "fontSize": 22,
        "fontWeight": "bold",
        "marginBottom": 54,
        "maxWidth": 277,
        "textAlign": "center",
      }
    }
  >
    We Are Sending You An SMS Code
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityLabel="btn-inputSMSCode-0"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F8F8F8",
          "borderColor": "#C4C4C4",
          "borderRadius": 5,
          "borderWidth": 1,
          "display": "flex",
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
          "width": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityLabel="btn-inputSMSCode-1"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F8F8F8",
          "borderColor": "#C4C4C4",
          "borderRadius": 5,
          "borderWidth": 1,
          "display": "flex",
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
          "width": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityLabel="btn-inputSMSCode-2"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F8F8F8",
          "borderColor": "#C4C4C4",
          "borderRadius": 5,
          "borderWidth": 1,
          "display": "flex",
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
          "width": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityLabel="btn-inputSMSCode-3"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F8F8F8",
          "borderColor": "#C4C4C4",
          "borderRadius": 5,
          "borderWidth": 1,
          "display": "flex",
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
          "width": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      blurOnSubmit={true}
      keyboardType="number-pad"
      maxLength={4}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "opacity": 0,
          "position": "absolute",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
